modules/heat_conduction/test/tests/interface_heating_mortar/constraint_joule_heating_dual_material.i

```diff
@@ -0,0 +1,277 @@
+    new_block_id = 4
+    new_block_name = 'interface_primary_subdomain'
+    input = interface_secondary_subdomain
+  []
+[]
+
+[Problem]
+  type = ReferenceResidualProblem
+  reference_vector = 'ref'
+  extra_tag_vectors = 'ref'
+[]
+
+[Variables]
+  [temperature]
+    initial_condition = 300.0
+  []
+  [temperature_interface_lm]
+    block = 'interface_secondary_subdomain'
+  []
+  [potential]
+  []
+  [potential_interface_lm]
+    block = 'interface_secondary_subdomain'
+  []
+[]
+
+[AuxVariables]
+  [interface_normal_lm]
+    order = FIRST
+    family = LAGRANGE
+    block = 'interface_secondary_subdomain'
+    initial_condition = 1.0
+  []
+[]
+
+[Kernels]
+  [HeatDiff_steel]
+    type = ADHeatConduction
+    variable = temperature
+    thermal_conductivity = steel_thermal_conductivity
+    extra_vector_tags = 'ref'
+    block = 'left_block'
+  []
+  [HeatDiff_aluminum]
+    type = ADHeatConduction
+    variable = temperature
+    thermal_conductivity = aluminum_thermal_conductivity
+    extra_vector_tags = 'ref'
+    block = 'right_block'
+  []
+  [electric_steel]
+    type = ADMatDiffusion
+    variable = potential
+    diffusivity = steel_electrical_conductivity
+    extra_vector_tags = 'ref'
+    block = 'left_block'
+  []
+  [electric_aluminum]
+    type = ADMatDiffusion
+    variable = potential
+    diffusivity = aluminum_electrical_conductivity
+    extra_vector_tags = 'ref'
+    block = 'right_block'
+  []
+[]
+
+[BCs]
+  [temperature_left]
+    type = ADDirichletBC
+    variable = temperature
+    value = 300
+    boundary = 'moving_block_left'
+  []
+  [temperature_right]
+    type = ADDirichletBC
+    variable = temperature
+    value = 300
+    boundary = 'fixed_block_right'
+  []
+
+  [electric_left]
+    type = ADDirichletBC
+    variable = potential
+    value = 0.0
+    boundary = moving_block_left
+  []
+  [electric_right]
+    type = ADDirichletBC
+    variable = potential
+    value = 3.0e-1
+    boundary = fixed_block_right
+  []
+[]
+
+[Constraints]
+  [thermal_contact]
+    type = ModularGapConductanceConstraint
+    variable = temperature_interface_lm
+    secondary_variable = temperature
+    primary_boundary = moving_block_right
+    primary_subdomain = interface_primary_subdomain
+    secondary_boundary = fixed_block_left
+    secondary_subdomain = interface_secondary_subdomain
+    gap_flux_models = 'closed_temperature'
+  []
+  [electrical_contact]
+    type = ModularGapConductanceConstraint
+    variable = potential_interface_lm
+    secondary_variable = potential
+    primary_boundary = moving_block_right
+    primary_subdomain = interface_primary_subdomain
+    secondary_boundary = fixed_block_left
+    secondary_subdomain = interface_secondary_subdomain
+    gap_flux_models = 'closed_electric'
+  []
+  [interface_heating]
+    type = InterfaceJouleHeatingConstraint
+    potential_lagrange_multiplier = potential_interface_lm
+    variable = temperature
+    secondary_variable = temperature
```

Comment:
Is this pattern variable/secondary_variable different from the existing mortar constraint above? I.e. in the new constraint both are primal variables whereas in the existing one it was primal variable and Lagrange multiplier.

Reply:
Or was this just a requirement to avoid an error from FEProblem and it does not matter what you input there?

Reply:
Both to avoid the error from FEProblem and to get the constraint contributions to add the heat source to the temperature field variable. When I used a dummy lagrange multiplier variable the temperature at the interface on the primary side did not match the analytical solution. 

I don't remember if I tried setting the primary variable to temperature and using a dummy/ placeholder lagrange multiplier as the variable. Would you like me to try that pattern for consistency with the other mortar blocks?

Reply:
Yeah, that'd be good if you push again. This is an inconsistency in the design, which doesn't happen in other mortar objects without LMs (e.g. penalty equal value). I'll look into it.

Reply:
this variation (from the single material insulated case that I have an analytical solution for) 
```
  [interface_heating]
    type = InterfaceJouleHeatingConstraint
    potential_lagrange_multiplier = potential_interface_lm
    variable = heat_source_interface_lm ## dummy lm variable created just for this spot
    primary_variable = temperature
    secondary_variable = temperature
    primary_electrical_conductivity = aluminum_electrical_conductivity
    secondary_electrical_conductivity = aluminum_electrical_conductivity
    primary_boundary = moving_block_right
    primary_subdomain = interface_primary_subdomain
    secondary_boundary = fixed_block_left
    secondary_subdomain = interface_secondary_subdomain
  []
```
fails with ```
Linear solve did not converge due to DIVERGED_PC_FAILED iterations 0
                 PC failed due to FACTOR_NUMERIC_ZEROPIVOT 
Nonlinear solve did not converge due to DIVERGED_FNORM_NAN iterations 0
```

while this version, that reuses the lagrange multiplier from the electrical potential contact does match the analytical solution:
```
  [interface_heating]
    type = InterfaceJouleHeatingConstraint
    potential_lagrange_multiplier = potential_interface_lm
    variable = potential_interface_lm ## reusing the electric potential lm
    primary_variable = temperature
    secondary_variable = temperature
    primary_electrical_conductivity = aluminum_electrical_conductivity
    secondary_electrical_conductivity = aluminum_electrical_conductivity
    primary_boundary = moving_block_right
    primary_subdomain = interface_primary_subdomain
    secondary_boundary = fixed_block_left
    secondary_subdomain = interface_secondary_subdomain
  []
  ```

Reply:
Should I rewrite the constraint class to use the `variable` parameter to supply the electric potential lm variable, instead of having a separate input file parameter for it?

Reply:
> would you point me to a penalty equal value test with the pattern you would like to see here? Then I will know what to make work--or at least what my goal is ;-)

I think all the tests with `PenaltyEqualValueConstraint` don't need to add that dummy variable. For example, `continuity_penalty_sphere_hex8.i`.

Reply:
> Should I rewrite the constraint class to use the variable parameter to supply the electric potential lm variable, instead of having a separate input file parameter for it?

That could be a way around it. But I think we'd be cheating since that constraint doesn't need to add a Lagrange multiplier and is only using another one to compute an equation (but that object could potentially get that information from a non-LM variable)

Reply:
Just to close the loop on this one, for future reference. The culprit appeared to be the integrity checks (kernel coverage) that, one way or another, assume that mortar constraints create or have Lagrange multipliers associated with it. This is no good in general, e.g.  for penalty approaches or for cases like the one in this PR in which we just apply a generalized force in the mortar sense without solving any constraint equation. Hopefully we can get something merge that'll fix it: https://github.com/idaholab/moose/pull/22125

Reply:
Until a fix makes it to the code, disabling kernel coverage should do the trick.